src/app-config.ts

```diff
@@ -29,16 +30,20 @@ if (!process.env.REACT_APP_MOCK && process.env.NODE_ENV !== 'test') {
     throw new Error('REACT_APP_TRAJECTORY_SERVER env variable is needed but not defined');
   }
 
+  const ws = new WebSocket(trajServer);
+
   appConfig = {
     transportFactory: () => SossTransport.connect(sossNodeName, sossServer, token),
-    trajectoryManagerFactory: () => DefaultTrajectoryManager.create(trajServer),
+    trajectoryManagerFactory: () => DefaultTrajectoryManager.create(ws),
     appResources: ResourceManager.getResourceConfigurationFile(),
+    trajectoryWebSocket : ws,
```

Comment:
> open a separate connection

the backend iterates through connections to send negotiation updates so doing this would be doubling bandwidth sent

> trajectory manager to support negotation status 

if we do this it sets up a precedent where anything that needs the websocket gets shoved into the trajectory manager. 

The only thing that will need tending to in future is the onmessage function binding inNegotiationStatusManager's`startSubscription()`, for the case where some other class needs a callback to another message type. ATM we should be fine sticking to this.

Reply:
> the backend iterates through connections to send negotiation updates so doing this would be doubling bandwidth sent

Can you explain more on this, do you mean that it will push the negotiation updates to all existing connections?

The trajectory protocol doesn't support interleaved requests atm, it always assumes that the response that it receives will be for the most recent requests sent, sharing connections will cause it interpret another instance's request for its own. This is also the case for negotiation push messages, it will cause it to interpret a negotiation message pushed by the server as a response to another request.

imo, without a proper protocol that can support interleaved messages, sharing connection would just cause more problems down the road.

Reply:
> Can you explain more on this, do you mean that it will push the negotiation updates to all existing connections?

See https://github.com/osrf/rmf_schedule_visualizer/blob/master/rmf_schedule_visualizer/src/Server.cpp#L90
Note that this is a server->romidashboard flow. It's not the same trajectory request->respond protocol that was previously setup.

> imo, without a proper protocol that can support interleaved messages, sharing connection would just cause more problems down the road.

Yea I'm very rarely getting exceptions related to that. I see your point; ill switch to use a second one.

Reply:
use 2 websockets 3472808
issue https://github.com/osrf/romi-dashboard/issues/102